persist/fs/mmap.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package fs
+
+import (
+	"fmt"
+	"os"
+)
+
+type mmapFileDesc struct {
+	// file is the *os.File ref to store
+	file **os.File
+	// bytes is the []byte slice ref to store the mmap'd address
+	bytes *[]byte
+	// options specifies options to use when mmaping a file
+	options mmapOptions
+}
+
+type mmapOptions struct {
+	// read is whether to make mmap bytes ref readable
+	read bool
+	// write is whether to make mmap bytes ref writable
+	write bool
```

Comment:
Might be better to not support write at all considering it's not something we're going to use in production. E.g. concern: we'd have to be careful about `msync` calls to ensure memory writes are synced to the filesystem. Don't need to spend the time figuring this out considering we're not using it.

Reply:
Fair, sure I'll remove this.